pkg/mcpproxy/server.go

```diff
@@ -74,6 +84,21 @@ func createProxyClient(ctx context.Context, config *ServerConfig) (*mcp.ClientSe
 		}
 	} else {
 		cmd := exec.Command(config.Command, config.Args...)
+
+		// Set up environment variables
+		if config.Env != nil {
+			env, err := buildEnv(config.Env)
+			if err != nil {
+				return nil, fmt.Errorf("failed to build environment: %w", err)
+			}
+			cmd.Env = env
```

Comment:
Note: this is the main change
the rest is tests and support for variable expansion and support for default values, which we can remove to make this PR smaller…